### PR TITLE
feat: update getSignatureStatus api

### PR DESCRIPTION
**Pull request by jstarry**
Depends on https://github.com/solana-labs/solana/pull/9069

### CODE REVIEW

File: test/connection.test.js [hunk 1]
```diff
@@ -1066,24 +1095,47 @@ test('transaction', async () => {
...
+
+  if (typeof firstResponse.confirmations !== 'number') {
+    expect(typeof firstResponse.confirmations).toEqual('number');
+    return;
+  }
+
+  expect(firstResponse.confirmations).toBeGreaterThanOrEqual(
+    responseConfirmations,
+  );
+
```
Comment (CriesofCarrots) [inline test/connection.test.js]: What do you think about adding a test case here for `confirmations: null`? Should be able to loop on getSignatureStatus with commitment: 'max' to know when the transaction is rooted for the live test.

Seems like a pretty easy add, and it'd be nice to have that tested in an integration test somewhere, at least.

Comment (jstarry) [reply test/connection.test.js]: Yes, great call!
